include/netpacket/bluetooth.h

```diff
@@ -286,6 +271,13 @@ struct sockaddr_bt_s
   uint8_t      bt_channel; /* Channel identifier (CID) */
 };
 
+struct sockaddr_hci_s
+{
+  sa_family_t hci_family;
+  uint8_t     hci_dev;
+  uint8_t     hci_channel;
+};
+
```

Comment:
The HCI address structure is basically the same as Linux (struct sockaddr_hci):

https://elixir.bootlin.com/linux/latest/source/include/net/bluetooth/hci_sock.h#L37

But not the L2CAP address (struct sockaddr_l2):

https://elixir.bootlin.com/linux/latest/source/include/net/bluetooth/l2cap.h#L65

Does this matter?

Reply:
> Does this matter?

This really raises the general question of "How close should the NuttX Bluetooth socket interface match the Linux Bluetooth socket interface?" 

1. Should Linux code just drop in with no changes other than perhaps header file paths?
2. Or should re-use of Linux code require adjusting to NuttX naming conventions like sockaddr_hci -> sockaddr_hci_s?
3. Or are we just aiming to be similar in spirit with Linux so that the port is easy?

The first option might set the expectation too high.  I think the NuttX implementation will (and correctly should) always be lightweight compared to the Linux implementation.

Reply:
I'm kind of aiming somewhere in the middle.  I was also wondering if I should rename the struct or not, I think keeping the members the same is important.

Adapting L2CAP to the Linux addr interface is what I would also like, but I am not focusing too much on that yet.

This also brings up the question of how much existing compatibility is ok to break in the spirit of better aligning with the "standard".

Reply:
> This also brings up the question of how much existing compatibility is ok to break in the spirit of better aligning with the "standard".

I think we are okay to change whatever we need to.  The nuttx/ and apps/ repository code should all be internally consistent and if we break any compatibility with user applications, we need to document the breakage that they should expect in the ReleaseNotes along with how to fix the breakage.

But I agree with your original statement.  There really is no "standard" that we have to conform to.  I don't think it is our objective to fully re-implement the Linux Bluetooth stack, but to produce and embedded implementation that is in alignment with the Linux stack.

I don't expect that we will ever be fully functionally compatible with Linux.

Naming is "low hanging fruit", however.  It is easy to change names  and I don't think that any significant user incompatibilities should result from that.   But on the other hand, there are aesthetic benefits with being compatible with the NuttX coding standard too.  I can go either way.

Reply:
I added hci l2cap and rfcomm socket address types and updated the references in the code.  This breaks apps, but I have have a branch that makes the minor changes there.

Reply:
Ok this is rebased on top of the changes that were made to define the new socket address types.  So going forward this should be able to remain in sync with the CI system as the interfaces have been defined.